contrib/mesos/pkg/scheduler/fcfs.go

```diff
@@ -67,7 +70,22 @@ func (fps *fcfsPodScheduler) SchedulePod(r offers.Registry, unused SlaveIndex, t
 		if offer == nil {
 			return false, fmt.Errorf("nil offer while scheduling task %v", task.ID)
 		}
-		if fps.FitPredicate()(task, offer) {
+
+		// try to get a node, might fail if it's the first pod on this slave
+		nodeName := offer.GetHostname()
```

Comment:
it's only the type cast which makes this a bit complicated. I would prefer to leave this as it is.